### PR TITLE
completed signup and login functionality and added toast 

### PR DESCRIPTION
**Pull request by Developers-codz**
## Files to check

`src/Redux/Reducers/authSlice.js`
`src/Redux/store.js`
`src/components/authComponent/RequireAuth.js`
`src/components/authComponent/RestrictAuth.js`
`src/pages/Authpages/Login.jsx`
`src/pages/Authpages/SignUp.jsx`

## [Live](https://share-arts-git-auth-developers-codz.vercel.app/)

Thank you for reviewing my pr : )

### CODE REVIEW

File: src/components/authComponent/RequireAuth.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import {Navigate,Outlet,useLocation} from "react-router-dom"
+import { useSelector } from "react-redux";
+
+export const RequireAuth = () => {
+    const currentUser = useSelector((state)=>state.auth.currentUser)
+    console.log(currentUser._id)
```
Comment (abhisheksahani-nova) [inline src/components/authComponent/RequireAuth.js]: Remove console.log if not needed

Comment (Developers-codz) [reply src/components/authComponent/RequireAuth.js]: done, thanks for pointing that out

**Review by abhisheksahani-nova — APPROVED** [review]
Everything else looks good and subtle, good to merge

File: src/Redux/Reducers/authSlice.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+
+const authSlice = createSlice({
+    name:"auth",
+    initialState,
+    extraReducers(builder) {
+        builder.addCase(login.fulfilled,(state,action) =>{
+            localStorage.setItem("token",action.payload.encodedToken)
+            state.currentUser = action.payload.foundUser
+            SuccessToast("Login Successful")
```
Comment (Vishnu-Aithal) [inline src/Redux/Reducers/authSlice.js]: I think actions other than setting state should not be done in reducers,
you can move SuccessToast to thunk function 'try' block like line number 15 before returning data, and ErrorToast to 'catch' block like line number 28 before. 
Even managing locale storage is side Effect, that should also be moved to thunk

**Review by Vishnu-Aithal — APPROVED** [review]
Reducers function must be pure and only set new state, any other dispatching like toast can be done in thunk .
managin local storage is also a side effect, that should also be moved to thunk.
Everything else looks good to merge.

File: src/pages/Authpages/SignUp.jsx [hunk 3]
```diff
@@ -4,26 +4,81 @@ import {
...
+  const clickHandler = () => {
+    const {fullname,email,username,password} = userDetail;
+    console.log(userDetail,passMatch)
+    if(fullname && email && username && password !== "" && password === passMatch ){
+      dispatch(signup(userDetail))
+      setUserDetail(prev => ({...prev,fullname:"",email:"",username:"",password:""}))
+      setPassMatch("")
+    }
+    else {
+   console.log("please fill all the fields")
```
Comment (Vishnu-Aithal) [inline src/pages/Authpages/SignUp.jsx]: suggestion - this can be a toast

File: src/Redux/Reducers/authSlice.js (hunk 2 above)
Comment (Developers-codz) [reply src/Redux/Reducers/authSlice.js]: Okay, will do that

File: src/pages/Authpages/SignUp.jsx (hunk 3 above)
Comment (Developers-codz) [reply src/pages/Authpages/SignUp.jsx]: Yeah, done

**Pull request merged**